clang/runtime/dpct-rt/include/dpct/device.hpp

```diff
@@ -768,17 +786,18 @@ class dev_mgr {
 #endif
   }
   void check_id(unsigned int id) const {
+    std::lock_guard<std::recursive_mutex> lock(m_mutex);
```

Comment:
For read-only, maybe we can assume the safety.

Reply:
The existing implementation always gets the mutex lock before calling `check_id`. This mod is in line with the existing implementation.

Reply:
Agreed.